### PR TITLE
Pages Editor: implement Edit Task Form

### PR DESCRIPTION
**Pull request by shaunanoordin**
## PR Overview

Part of: **Pages Editor MVP** project and **FEM Lab** super-project
Follows #6973
Staging branch URL: https://pr-6981.pfe-preview.zooniverse.org/lab/1982/workflows/editor/3711?env=staging

This PR implements the Edit Task Form (the actual interesting bits inside the Edit Step/Page dialog).

- Currently supports editing: **Text Task** and **Single Question Task**
  - Deferred for another PR: Drawing Task
- Text Task: project owners can edit main text, help text, and is required y/n.
- Single Question Task: project owners can...
  - edit main text, help text and is required y/n
  - add, edit, and delete choices (Answers)
  - NOT choose the next Step/Page yet. That's for the next PR.

_Screenshot: user editing Page P0, Text Task T0_

image

_Screenshot: user editing Page P0, Single Question Task T0 with 3 choices_

image

### Status

~~WIP~~ Ready for review!

Do not merge until 6973 is merged. Currently targets pages-editor-pt11 for reviewing purposes; this will be re-targeted to master when ready.

### CODE REVIEW

**Comment by shaunanoordin** [comment]
## PR Update

We can now edit Single Question Tasks. This PR is now ready for review!

File: app/pages/lab-pages-editor/components/TasksPage/components/EditStepDialog/types/SingleQuestionTask.jsx [hunk 1]
```diff
@@ -0,0 +1,157 @@
...
+              type="checkbox"
+              checked={required}
+              onChange={(e) => {
+                setRequired(!!e?.target?.checked);
+              }}
+            />
+            <span>
+              Required
+            </span>
+          </label>
```
Comment (eatyourgreens) [inline app/pages/lab-pages-editor/components/TasksPage/components/EditStepDialog/types/SingleQuestionTask.jsx:110]: A11y tip: always use `for` with labels, as explicit labels are better supported by assistive tech, eg. Dragon Naturally Speaking.
https://github.com/w3c/aria-practices/issues/2870

File: app/pages/lab-pages-editor/components/TasksPage/components/EditStepDialog/types/SingleQuestionTask.jsx [hunk 2]
```diff
@@ -0,0 +1,157 @@
...
+        </div>
+      </div>
+      <div className="input-row">
+        <ul>
+          {answers.map(({ label, next }, index) => (
+            <li
+              aria-label={`Choice ${index}`}
```
Comment (eatyourgreens) [inline app/pages/lab-pages-editor/components/TasksPage/components/EditStepDialog/types/SingleQuestionTask.jsx:117]: This label should be on the text input, not the list item.

File: app/pages/lab-pages-editor/components/TasksPage/components/EditStepDialog/types/TextTask.jsx [hunk 3]
```diff
@@ -1,27 +1,78 @@
...
-          <button>Delete</button>
         </div>
+        {/* <button>Delete</button> */}
       </div>
-      <div>
-        <label>
-          <input type="checkbox" />
-          Required
+      <div className="input-row">
+        <label className="narrow">
```
Comment (eatyourgreens) [inline app/pages/lab-pages-editor/components/TasksPage/components/EditStepDialog/types/TextTask.jsx:50]: Add `htmlFor` to this label too.

File: app/pages/lab-pages-editor/components/TasksPage/TasksPage.jsx [hunk 4]
```diff
@@ -123,6 +135,8 @@ export default function TasksPage() {
           allTasks={workflow.tasks}
           step={workflow.steps[activeStepIndex]}
           stepIndex={activeStepIndex}
+          deleteTask={deleteTask}
+          updateTask={updateTask}
         />
```
Comment (kieftrav) [inline app/pages/lab-pages-editor/components/TasksPage/TasksPage.jsx:140]: Stylistic thing to put attributes in alphabetical order

File: app/pages/lab-pages-editor/components/TasksPage/components/EditStepDialog/EditStepDialog.jsx [hunk 5]
```diff
@@ -4,10 +4,17 @@ import PropTypes from 'prop-types';
...
 import CloseIcon from '../../../../icons/CloseIcon.jsx';
 
+const taskNames = {
+  'drawing': 'Drawing',
+  'single': 'Single Question',
+  'text': 'Text',
+}
+
 function EditStepDialog({
   allTasks = {},
```
Comment (kieftrav) [inline app/pages/lab-pages-editor/components/TasksPage/components/EditStepDialog/EditStepDialog.jsx:14]: this is getting really nit-picky and very personal style... But I like putting the primary / common noun first. For example, `tasksAll` and `taskUpdate`. This makes browsing through the code a bit easier to understand.

File: app/pages/lab-pages-editor/components/TasksPage/components/EditStepDialog/EditStepDialog.jsx [hunk 6]
```diff
@@ -28,7 +35,9 @@ function EditStepDialog({
     };
   });
 
-  const title = 'Create a (???) Task'
+  const firstTask = allTasks?.[taskKeys?.[0]]
```
Comment (kieftrav) [inline app/pages/lab-pages-editor/components/TasksPage/components/EditStepDialog/EditStepDialog.jsx:38]: Same here... I'd personally prefer `taskFirst` with `tasksAll`

File: app/pages/lab-pages-editor/components/TasksPage/components/EditStepDialog/EditStepDialog.jsx [hunk 7]
```diff
@@ -62,13 +71,21 @@ function EditStepDialog({
               key={`editTaskForm-${taskKey}`}
```
Comment (kieftrav) [inline app/pages/lab-pages-editor/components/TasksPage/components/EditStepDialog/EditStepDialog.jsx:71]: camelCase and kebab-case mixing

**Review by kieftrav — APPROVED** [review]
LGTM - some stylistic suggestions on top of Jim's comments

**Comment by shaunanoordin** [comment]
Thanks Jim and Travis!

Required follow up once I get this PR merge train done:
- fix accessibility issues
- review code style
- 🐛 ❗ Examine **can't rearrange certain Pages via key** issue
- (from 6973) implement DEFAULT_HANDLER

Merging this first and then moving to 6991

**Pull request merged**